source/Animate.h

```diff
@@ -0,0 +1,67 @@
+/* Animate.h
+Copyright (c) 2023 by Rian Shelley
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef ANIMATE_H_INCLUDED
+#define ANIMATE_H_INCLUDED
+
+// Smoothly change a variable from one value to another. Used to smooth out
+// scrolling and panning.
+template <typename VarT>
+class Animate
+{
+public:
+	virtual ~Animate() = default;
+	// Set the next target value of this variable, linearly interpolated using
+	// steps frames.
+	virtual void Set(const VarT & current, int steps = 5)
+	{
```

Comment:
> Definitions for template functions should be outside the class, look at Date.h for an example.

I'm confused. `Date.h` is not a template class. 

With template classes, you can't break your class into .h and .cxx files, so there isn't much point in separating the definition from the declaration.

Reply:
> > Definitions for template functions should be outside the class, look at Date.h for an example.
> 
> I'm confused. `Date.h` is not a template class.
> 
> With template classes, you can't break your class into .h and .cxx files, so there isn't much point in separating the definition from the declaration.

D'oh, I meant `Dialog.h`

Reply:
> I think Animate.h is especially redundant

I have future PRs queued up that use `Animate` in lots of other contexts where `ScrollVar` does not apply. Specifically, `ScrollVar` requires a well defined `<` operator, where `Animate` only requires addition and integer division. 

This allows `Animate` to be used with the `Point` class, and it can be used to scroll map coordinates into view. That is for a future PR though.